astropy/io/fits/diff.py

```diff
@@ -29,7 +29,7 @@
 from .card import Card, BLANK_CARD
 from .header import Header
 # HDUList is used in one of the doctests
-from .hdu.hdulist import fitsopen, HDUList  # pylint: disable=W0611
```

Comment:
When the test runner runs doctests it runs them with the module locals of the module they were written in, so that's why it works.  But better to put the necessary imports in the doctests too so that users know what they need to import and from where.